From aa580ece445b3ca02d09c036be5b1d28959a77af Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Sat, 14 May 2022 22:57:47 +0800
Subject: [PATCH 41/58] Revert "drm/amd/display: Add affected crtcs to atomic
 state for dsc mst unplug"

This reverts commit 9c384e1afa552c94b5684036d5a26fa6dc11c37e.
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index e828f9414ba2c..b653646952196 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -8615,13 +8615,10 @@ static int dm_update_plane_state(struct dc *dc,
 static int add_affected_mst_dsc_crtcs(struct drm_atomic_state *state, struct drm_crtc *crtc)
 {
 	struct drm_connector *connector;
-	struct drm_connector_state *conn_state, *old_conn_state;
+	struct drm_connector_state *conn_state;
 	struct amdgpu_dm_connector *aconnector = NULL;
 	int i;
-	for_each_oldnew_connector_in_state(state, connector, old_conn_state, conn_state, i) {
-		if (!conn_state->crtc)
-			conn_state = old_conn_state;
-
+	for_each_new_connector_in_state(state, connector, conn_state, i) {
 		if (conn_state->crtc != crtc)
 			continue;
 
-- 
2.27.0

